src/components/VmDetails/cards/NicsCard/index.js

```diff
@@ -1,96 +1,213 @@
+    }
+
+    if (Object.values(changes).length > 0) {
+      this.setState(changes)
+    }
+  }
+
+  onCreateConfirm (nic) {
+    console.log('action - create', nic)
+    this.props.addNic({ nic })
+  }
+
+  onEditConfirm (nic) {
+    console.log('action - edit', nic)
+    this.props.editNic({ nic })
+  }
+
+  onDeleteConfirm (nicId) {
+    console.log('action - delete', nicId)
+    this.props.deleteNic({ nicId })
+  }
+
+  render () {
+    const { vm, vnicProfiles, onEditChange } = this.props
+
+    const canEditTheCard =
+      vm.get('canUserEditVm', true) && // TODO: default to true until PR#784 is merged
+      vm.getIn(['pool', 'id']) === undefined
+
+    const vmStatus = vm.get('status')
+    const canCreateNic = true // TODO: True for admins, may need permission checks for users
+
+    const showNicIPs = vm.get('status') === 'up'
+    const nicList = vm.get('nics')
+      .sort((a, b) => a.get('name').localeCompare(b.get('name')))
+      .map(nic => ({
+        id: nic.get('id'),
+        name: nic.get('name'),
+        plugged: nic.get('plugged'),
+        linked: nic.get('linked'),
+        interface: nic.get('interface'),
+        vnicProfile: {
+          id: nic.getIn(['vnicProfile', 'id']),
+          name: vnicProfiles.getIn([nic.getIn(['vnicProfile', 'id']), 'name']),
+          network: vnicProfiles.getIn([nic.getIn(['vnicProfile', 'id']), 'network', 'name']),
+        },
+        ipv4: showNicIPs ? nic.get('ipv4').toJS() : [],
+        ipv6: showNicIPs ? nic.get('ipv6').toJS() : [],
+        canEdit: true, // TODO: True for admins, may need permission checks for users
+        canDelete: vmStatus === 'down' || !nic.get('plugged'),
+      }))
+      .toJS()
+
+    return (
+      <BaseCard
+        icon={{ type: 'pf', name: 'network' }}
+        title={msg.nic()}
+        editTooltip={msg.nicCardEditTooltip({ vmId: vm.get('id') })}
+        editable={canEditTheCard}
+        itemCount={vm.get('nics').size}
+        onStartEdit={() => { onEditChange(true) }}
+        onCancel={() => { onEditChange(false) }}
+        onSave={() => { onEditChange(false) }}
+      >
+        {({ isEditing }) =>
+          <Grid className={style['nics-container']}>
+            { isEditing && canCreateNic &&
+              <Row key={`${vm.get('id')}-nic-add`}>
+                <Col>
+                  <NicEditor
+                    nextNicName={this.state.nextNicName}
+                    vnicProfileList={this.state.filteredVnicList}
+                    onSave={this.onCreateConfirm}
+                    trigger={
+                      <a href='#' id='nic-create-action'>
+                        <Icon className={style['add-icon']} type='fa' name='plus' />
+                        <span className={style['add-text']} >{msg.nicActionCreateNew()}</span>
+                      </a>
+                    }
+                  />
+                </Col>
+              </Row>
+            }
+
+            { nicList.length === 0 &&
```

Comment:
I think better use `!nicList.length`, instead of `nicList.length === 0`, it is more clear, and in case of `nicList.length === undefined` it would have better fault tolerance.

Reply:
Since `nicList` is created in the `render` function, it will always be an array here.  The `undefined` case can be ignored.

The `nicList.length === 0` is explicit in saying "when `nicList` has 0 nics".  Pair that with the `nicList.length > 0` check in the next block, and I would prefer to keep the check the way it is.